### PR TITLE
Added in feature for missed fullscreen mode

### PR DESCRIPTION
**Pull request by mike-st**
also adds transcript functionality

### CODE REVIEW

File: properties.schema [hunk 1]
```diff
@@ -104,6 +106,15 @@
           "validators": [],
           "help": "This parameter lets you use a YouTube player that does not show a YouTube logo. Set to true to prevent the YouTube logo from displaying in the control bar. Note that a small YouTube text label will still display in the upper-right corner of a paused video when the user's mouse pointer hovers over the player."
         },
+        "_allowFullscreen": {
+          "type":"boolean",
+          "required":false,
+          "default": "true",
+          "title": "Allow Fullscreen View",
+          "inputType": {"type": "Boolean", "options": [true, false]},
```
Comment (moloko) [inline properties.schema:114]:
```suggestion
          "inputType": "Checkbox",
```

File: properties.schema [hunk 2]
```diff
@@ -104,6 +106,15 @@
...
+          "validators": [],
+          "help": "This parameter lets you use the fullscreen function on a Youtube video. When press the video will fill the screen."
```
Comment (moloko) [inline properties.schema:116]: typo "When press the video"... however, I don't think you really need help for this entry, the title's descriptive enough

File: properties.schema [hunk 3]
```diff
@@ -140,6 +151,84 @@
...
+      "type":"object",
+      "required":false,
+      "title": "Transcript Options",
+      "properties":{
+        "_setCompletionOnView": {
+          "type":"boolean",
+          "required":false,
+          "default": true,
+          "title": "Set Completion",
+          "inputType": {"type": "Boolean", "options": [true, false]},
```
Comment (moloko) [inline properties.schema:165]:
```suggestion
          "inputType": "Checkbox",
```

File: properties.schema [hunk 4]
```diff
@@ -140,6 +151,84 @@
...
+          "validators": [],
+          "help": "This tells to set this video to complete when the inline transcript is shown"
```
Comment (moloko) [inline properties.schema:167]:
```suggestion
          "help": "Whether to have this component mark as completed when the transcript is shown."
```

File: properties.schema [hunk 5]
```diff
@@ -140,6 +151,84 @@
...
+        },
+        "_inlineTranscript": {
+          "type":"boolean",
+          "required":false,
+          "default": false,
+          "title": "Show Inline",
+          "inputType": {"type": "Boolean", "options": [true, false]},
```
Comment (moloko) [inline properties.schema:174]:
```suggestion
          "inputType": "Checkbox",
```

File: properties.schema [hunk 6]
```diff
@@ -140,6 +151,84 @@
...
+          "validators": []
+        },
+        "_externalTranscript": {
+          "type":"boolean",
+          "required":false,
+          "default": false,
+          "title": "Show Link",
+          "inputType": {"type": "Boolean", "options": [true, false]},
```
Comment (moloko) [inline properties.schema:182]:
```suggestion
          "inputType": "Checkbox",
```

File: properties.schema [hunk 7]
```diff
@@ -140,6 +151,84 @@
...
+    "_transcript": {
+      "type":"object",
+      "required":false,
+      "title": "Transcript Options",
+      "properties":{
+        "_setCompletionOnView": {
+          "type":"boolean",
+          "required":false,
+          "default": true,
+          "title": "Set Completion",
```
Comment (moloko) [inline properties.schema:164]:
```suggestion
          "title": "Trigger completion?",
```

File: properties.schema [hunk 8]
```diff
@@ -43,7 +45,7 @@
       "title": "Set completion of Video on",
       "inputType": {"type": "Select", "options":["inview", "play", "ended"]},
       "validators": ["required"],
-      "help": "This tells Adapt when to set this video to complete"
+      "help": "Tells when to set this video to complete"
```
Comment (moloko) [inline properties.schema:48]:
```suggestion
      "help": "Which event should trigger completion of this component."
```

File: templates/youtube.hbs [hunk 9]
```diff
@@ -1,17 +1,44 @@
...
-	      <a href="{{_media.transcriptLink}}" target="_blank" class="media-transcript">
-	        {{#if _media.transcriptText}}
-	          {{_media.transcriptText}}
-	        {{else}}
-	          {{_media.transcriptLink}}
-	        {{/if}}
-	      </a>
-	    </div>
-	    {{/if}}
+		<iframe width="640px" height="360px" style="width:100%;" src="{{_media._source}}?&autoplay={{#if _media._autoplay}}1{{else}}0{{/if}}&rel={{#if _media._showRelated}}1{{else}}0{{/if}}&loop={{#if _loop}}1{{else}}0{{/if}}&modestbranding={{#if _media._modestBranding}}1{{else}}0{{/if}}&color={{#if _media._progressColor}}{{_media._progressColor}}{{/if}}&controls=3&enablejsapi=1&playsinline=1&showinfo=0&iv_load_policy={{#if _media._showAnnotations}}1{{else}}3{{/if}}" frameborder="0" {{#if _media._allowFullscreen}}allowfullscreen="allowfullscreen"{{/if}}></iframe>
```
Comment (moloko) [inline templates/youtube.hbs:4]: the attribute `allowfullscreen` should have the value `"true"` - source: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#Attributes

```suggestion
		<iframe width="640px" height="360px" style="width:100%;" src="{{_media._source}}?&autoplay={{#if _media._autoplay}}1{{else}}0{{/if}}&rel={{#if _media._showRelated}}1{{else}}0{{/if}}&loop={{#if _loop}}1{{else}}0{{/if}}&modestbranding={{#if _media._modestBranding}}1{{else}}0{{/if}}&color={{#if _media._progressColor}}{{_media._progressColor}}{{/if}}&controls=3&enablejsapi=1&playsinline=1&showinfo=0&iv_load_policy={{#if _media._showAnnotations}}1{{else}}3{{/if}}" frameborder="0" {{#if _media._allowFullscreen}}allowfullscreen="true"{{/if}}></iframe>
```

File: js/adapt-youtube.js [hunk 10]
```diff
@@ -88,6 +91,29 @@ define(function(require) {
                 }
             }
         },
+        
+        onToggleInlineTranscript: function(event) {
+            if (event) event.preventDefault();
+            var $transcriptBodyContainer = this.$(".youtube-inline-transcript-body-container");
+            var $button = this.$(".youtube-inline-transcript-button");
```
Comment (moloko) [inline js/adapt-youtube.js:101]: You've got [this bug](https://github.com/adaptlearning/adapt_framework/issues/2062) here... you can fix by doing this:
```suggestion
            var $buttonText = this.$(".youtube-inline-transcript-button .transcript-text-container");
```
then changing the instances of `$button` on L105 and L111 to `$buttonText`

File: js/adapt-youtube.js [hunk 11]
```diff
@@ -88,6 +91,29 @@ define(function(require) {
...
+
+            if ($transcriptBodyContainer.hasClass("inline-transcript-open")) {
+                $transcriptBodyContainer.slideUp(function() {
```
Comment (moloko) [inline js/adapt-youtube.js:104]: You've got [this bug](https://github.com/adaptlearning/adapt_framework/issues/1718) here, you can fix by doing:
```suggestion
                $transcriptBodyContainer.stop(true, true).slideUp(function() {
```

File: js/adapt-youtube.js [hunk 12]
```diff
@@ -88,6 +91,29 @@ define(function(require) {
...
+                    $(window).resize();
+                });
+                $transcriptBodyContainer.removeClass("inline-transcript-open");
+                $button.html(this.model.get("_transcript").inlineTranscriptButton);
+            } else {
+                $transcriptBodyContainer.slideDown(function() {
```
Comment (moloko) [inline js/adapt-youtube.js:110]:
```suggestion
                $transcriptBodyContainer.stop(true, true).slideDown(function() {
```

**Review by moloko — CHANGES_REQUESTED** [review]
thanks for doing this, few minor changes needed - have done github 'suggestions' so they should be quick to implement

**Comment by moloko** [comment]
@mike-st have resolved the conflicts for you... was there a reason you set the FW dependency to a min of 2.0.9? if so, let me know - happy to change this back.

**Comment by mike-st** [comment]
Hi @moloko Thanks for the updates and fixes I will take a look at it soon. Makes sense from what I can tell so far. Appreciate your help. Framework 2.0.9 is the oldest one we work with so I just set it to that and above as we have a couple of instances running different framework versions. You can omit that one.

**Comment by moloko** [comment]
@mike-st if you want to leave the FW dependency as-is that's fine, I can always change it in our repo after merging the PR. Whatever's easiest for you.

**Pull request closed without merging**